src/hotspot/os/windows/os_windows.inline.hpp

```diff
@@ -49,12 +49,16 @@ inline void os::map_stack_shadow_pages(address sp) {
   // If we decrement stack pointer more than one page
   // the OS may not map an intervening page into our space
   // and may fault on a memory access to interior of our frame.
+  address original_sp = sp;
   const size_t page_size = os::vm_page_size();
   const size_t n_pages = StackOverflow::stack_shadow_zone_size() / page_size;
   for (size_t pages = 1; pages <= n_pages; pages++) {
     sp -= page_size;
     *sp = 0;
   }
+  StackOverflow* state = JavaThread::current()->stack_overflow_state();
+  assert(original_sp > state->shadow_zone_safe_limit(), "");
```

Comment:
Can you print the values if the assert fails please.

Reply:
Done.